### ISSUE TITLE
Style: "text-align: center" breaks Flutter embedding

### ISSUE THREAD

**Issue by jacobsimionato** [flutter/flutter#127640]
Based on Google bug b/284379634 which has more context.

When you embed Flutter in a position in the DOM with `text-align: center` (and probably any alignment other than left), it makes the Flutter canvas get positioned with the leftmost edge in the center of the wrapper. The flutter-view is positioned correctly, but not the canvas.

The fix is to change the style back to `text-align: left` around the actual Flutter embedding e.g. with a div like:
```
<div style="text-align: center;">
  Some centered text here potentially.

  <div id="flutter-target" style="text-align: left;"></div>
  </div>
</div>
```

Example of problem:
![Screenshot 2023-05-26 at 1 09 18 pm](https://github.com/flutter/flutter/assets/6869388/9589ec8b-a0be-47ce-99f4-400ee8481c24)

Example of fix:
![Screenshot 2023-05-26 at 1 10 11 pm](https://github.com/flutter/flutter/assets/6869388/a7879ee3-3945-4cc9-be07-a36c15b05616)

Maybe we can consider resetting styles like this that break Flutter rendering inside the Flutter-owned DOM? Though I know David mentioned it might be too hard to commit to resetting everything that could possibly matter.

**Comment by ditman** [comment]
We may be able to use: `all: initial;` as the first style from on the flutter host element so we undo anything that comes from the outside (except... with enough specificity some of those styles could still be overridden!)

**Comment by mdebbar** [comment]
@ditman we might be able to get away with setting `all: initial;` as an inline style which [always wins](https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity#inline_styles) the specificity race.

**Comment by ditman** [comment]
@mdebbar yeah, that conflicts a little with "fixing" this: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/style-src#unsafe_inline_styles, I was looking at ways to stop using inline styles, or make them "not "unsafe""

**Comment by ditman** [comment]
(We might need to add a flutter.js parameter to have a nonce that the user passes into the engine, so we can "sign" our inline stuff with that value)
